core/src/main/scala/akka/persistence/jdbc/schema/scaladsl/SchemaUtils.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2014 - 2019 Dennis Vriend <https://github.com/dnvriend>
+ * Copyright (C) 2019 - 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.jdbc.schema.scaladsl
+
+import java.sql.Statement
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.dispatch.Dispatchers
+import akka.persistence.jdbc.db.SlickDatabase
+import akka.persistence.jdbc.db.SlickExtension
+import slick.jdbc.H2Profile
+import slick.jdbc.JdbcBackend.Database
+import slick.jdbc.MySQLProfile
+import slick.jdbc.OracleProfile
+import slick.jdbc.PostgresProfile
+import slick.jdbc.SQLServerProfile
+
+object SchemaUtils {
+
+  private[jdbc] sealed trait SchemaType {
+    def schema: String
+    def separator: String
+    def matchingProfile: String
+  }
+
+  private[jdbc] object SchemaType {
+    case object Postgres extends SchemaType {
+      val schema = "schema/postgres/postgres-schema.sql"
+      val separator = ";"
+      val matchingProfile = "slick.jdbc.PostgresProfile$"
+    }
+    case object H2 extends SchemaType {
+      val schema = "schema/h2/h2-schema.sql"
+      val separator = ";"
+      val matchingProfile = "slick.jdbc.H2Profile$"
+    }
+    case object MySQL extends SchemaType {
+      val schema = "schema/mysql/mysql-schema.sql"
+      val separator = ";"
+      val matchingProfile = "slick.jdbc.MySQLProfile$"
+    }
+    case object Oracle extends SchemaType {
+      val schema = "schema/oracle/oracle-schema.sql"
+      val separator = "/"
+      val matchingProfile = "slick.jdbc.OracleProfile$"
+    }
+    case object SqlServer extends SchemaType {
+      val schema = "schema/sqlserver/sqlserver-schema.sql"
+      val separator = ";"
+      val matchingProfile = "slick.jdbc.SQLServerProfile$"
+    }
+  }
+
+  def dropAndCreate(implicit actorSystem: ActorSystem): Future[Done] = {
```

Comment:
Make it compatible with the new actors API out-of-the-box:
```suggestion
  def dropAndCreate(implicit actorSystem: ClassicActorSystemProvider): Future[Done] = {
```